BotModules/CommanderInChief/commander-in-chief.js

```diff
@@ -37,28 +42,85 @@ function registerHelp() {
 // Next command etc...
 //
 //  - msg: the message object from discord https://discord.js.org/#/docs/main/stable/class/Message
-//  TODO - allow to discriminate between different command contexts
-//  TODO - Show information about arguments
-function help(msg){
+function help(msg, args){
   var reply = `Looks like ${msg.member} is an idiot and needs guidence on the commands: \n\n`;
-  commands.forEach(command => {
-    reply += "Command: " + command.keyword + `\n`;
-    reply += command.helptext + `\n\n`
-  });
+  var filterContext = args.raw[1] ? args.raw[1].toUpperCase() : undefined;
+
+  var contexts = Array.from(new Set(commands.map(command => command.context))).sort();
+
+  switch(filterContext){
+    case 'ALL':
+      printAll();
+      break;
+    case 'LIST':
+      printListOfContexts();
+      break;
+    default:
+      printContext(filterContext);
```

Comment:
I think we should add a check against existing contexts and if the user provided context does not exist print either the standard /help or /help list. The output of a nonexistent context right now is:

`
232@@#@#@
=====
`

Reply:
Opps fixed in 8956031dab6a1252d5aa0241e88522f6c953fe77

Should output a list of supported contexts now